sys/color/color.c

```diff
@@ -27,51 +27,49 @@
 
 void color_rgb2hsv(color_rgb_t *rgb, color_hsv_t *hsv)
 {
-    float rd, gd, bd, delta;
+    float rd, gd, bd, delta, min, max;
+    int q, imax, imin;
 
     /* norm RGB colors to the range [0 - 1.0] */
-    rd = (float)rgb->r / 255.0f;
-    gd = (float)rgb->g / 255.0f;
-    bd = (float)rgb->b / 255.0f;
-
-    /* find value as maximum of the three colors */
-    if (rd >= gd) {
-        hsv->v = (rd >= bd) ? rd : bd;
+    rd = (float)rgb->r * (1 / 255.0f);
+    gd = (float)rgb->g * (1 / 255.0f);
+    bd = (float)rgb->b * (1 / 255.0f);
+
+    /* find maximum and minimum and quadrant of the three colors*/
+    imax = rgb->r;
+    max = rd;
+    q = 0;
+    if ( rgb->g > imax ) {
+        imax = rgb->g;
+        max = gd;
+        q = 1;
     }
-    else {
-        hsv->v = (gd >= bd) ? gd : bd;
+    if ( rgb->b > imax ) {
+        imax = rgb->b;
+        max = bd;
+        q = 2;
     }
+    /* value is maximum*/
+    hsv->v = max;
 
     /* compute delta from value and minimum of the three RGB colors */
-    if (rd <= gd) {
-        delta = (rd <= bd) ? (hsv->v - rd) : (hsv->v - bd);
-    }
-    else {
-        delta = (gd <= bd) ? (hsv->v - gd) : (hsv->v - bd);
-    }
+    imin = rgb->r, min = rd;
+    imin = (rgb->g < imin) ? (min = gd , rgb->g) : imin;
```

Comment:
```suggestion
    imin = rgb->r;
    min = rd;
   if (rgb->g < imin) {
       imin = rgb->g;
       min = gd;
   }
```

Reply:
don't you like the comma operator?

Reply:
No. Many people have to look up its meaning again to understand the code ;-)